src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java

```diff
@@ -192,21 +194,21 @@ private void computeTelemetrySettings() {
     }
 
     private void reportTelemetry() {
-        TelemetryMessageBuilder.ActionMessage telemetry = TelemetryService.instance().getBuilder().action(TelemetryService.NAME_PREFIX_MISC + "login");
+        TelemetryMessageBuilder.ActionMessage telemetry = instance().getBuilder().action(NAME_PREFIX_MISC + "login");
         try {
             ClusterInfo info = ClusterHelper.getClusterInfo(client);
             telemetry.property(KUBERNETES_VERSION, info.getKubernetesVersion());
             telemetry.property(IS_OPENSHIFT, Boolean.toString(info.isOpenshift()));
             telemetry.property(OPENSHIFT_VERSION, info.getOpenshiftVersion());
-            telemetry.send();
+            asyncSend(telemetry);
         } catch (RuntimeException e) {
             // do not send telemetry when there is no context ( ie default kube URL as master URL )
             try {
                 //workaround to not send null values
                 if (e.getMessage() != null) {
-                    telemetry.error(e).send();
+                    asyncSend(telemetry.error(e));
                 } else {
-                    telemetry.error(e.toString()).send();
+                    asyncSend(telemetry.error(e));
```

Comment:
should be 
```suggestion
                    asyncSend(telemetry.error(e.toString()));
```

Reply:
@sbouchet why?

Reply:
because above is already sending the event. this was a if/then clause to avoid sending null message in telemetry.

Reply:
this should be fixed in telemetry then.

Reply:
thus, no need to have this if/else anymore, and we need an issue/fix on telemetry first.